src/components/Button.tsx

```diff
@@ -0,0 +1,81 @@
+import { ButtonHTMLAttributes } from 'react';
+import { css, styled } from 'styled-components';
+import { ColorType, designSystem } from '../styles/designSystem';
+
+type ButtonType = 'container' | 'outline' | 'ghost' | 'circle';
+
+type ButtonPropsWithColor = {
+  children?: React.ReactNode;
+  styledType: Exclude<ButtonType, 'ghost'>;
+  color: ColorType;
+} & ButtonHTMLAttributes<HTMLButtonElement>;
+
+type ButtonPropsGhost = {
+  children?: React.ReactNode;
+  styledType: 'ghost';
+} & ButtonHTMLAttributes<HTMLButtonElement>;
+
+type ButtonProps = ButtonPropsWithColor | ButtonPropsGhost;
```

Comment:
type이 ghost일 때는 color prop을 허용하고 싶지 않아서 타입을 나누신 건가요?
현재 styledType ghost 이면서 color 넘겨줄 때 에러가 발생하지 않아서
1. 타입을 더 엄격하게 관리해보거나
2. 그냥 합쳐서 관리해도 될 것 같습니당

개인적으로는 2로 해도 무방할 것 같아요.

그리고 children이 없어도 되는 게 아니라면 required로 지정해주세요 !

```suggestion
type ButtonProps = {
  children?: React.ReactNode;
  type: ButtonType;
  color?: ColorType;
} & ButtonHTMLAttributes<HTMLButtonElement>;
```

Reply:
넵 저는 ghost 속성일 때 color를 사용하지 않는다면 굳이 props를 받을 필요도 없다고 생각해서 ghost일 때는 color를 안 받기 위해 두 가지 타입으로 분리했습니다.


우선 제 생각으로 최대한 구현한 부분이고 더 엄격하게 한다면 어떻게 할 수 있을지 잘 모르겠네요😅

Reply:
네네 그런 것이라면 ghost이면서 color를 넘겼을 때 타입에러가 발생하도록 수정이 필요할 것 같습니다!

Reply:
제네릭을 이용하면 될 것 같은데 아직은 잘 생각이 안 나네요
제 스스로에게 숙제로 남기고 차차 수정해 보겠습니다!!